wrangler-api/src/main/java/io/cdap/wrangler/api/WranglerMetric.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.api;
+
+import io.cdap.cdap.api.metrics.MetricType;
+
+import java.util.Map;
+
+/**
+ * Represents generic metadata information for any metric that is emitted in Wrangler
+ */
+public class WranglerMetric {
```

Comment:
nit: `EntityMetric` would make more sense since metrics are emitted by an entity which implements `EntityWithMetrics` interface

Reply:
+1, I would call this EntityMetricDef as it is confusing between EntityMetric vs EntityWithMetric. Also EntityWithMetric can be EntityMetric.

Reply:
Renamed on further discussion to EntityMetricDef and EntityMetrics